mongo/collection.go

```diff
@@ -748,6 +762,12 @@ func aggregate(a aggregateParams) (*Cursor, error) {
 	}
 
 	sess := sessionFromContext(a.ctx)
+	// Always close any created implicit sessions if aggregate returns an error.
+	defer func() {
+		if err != nil && sess != nil {
+			closeImplicitSession(sess)
+		}
```

Comment:
To always get the returned error, `err` has to reference a named return value. In the current code, `err` references the variable returned from `transformAggregatePipeline`, but may not reference error variables created in different scopes (e.g. the error returned from `transformValue` in the `if ao.Hint != nil {}` block).

Update the `aggregate` function signature to use named return values. E.g.
```go
func aggregate(a aggregateParams) (cur *Cursor, err error)

Reply:
🤦  Didn't actually _name_ the named return values lol. Done.